ImmunityReborn/Assets/_TeamAmnesia/Scripts/GameOverScreen.cs

```diff
@@ -22,8 +22,6 @@ public void Setup( float durationAlive )
 
     public void RestartGameButton()
     {
-        // Save the current level index for softcore
-        PlayerPrefs.SetInt("SCCurrentLevel", FindFirstObjectByType<LevelManager>().CurrentLevelIndex);
```

Comment:
Not good to do it here, instead I only set the current level on player state machine handle die.